### PR TITLE
Add missing docs for Task API

### LINKED ISSUE #105985: Port System.Threading.Tasks docs for .NET 9 APIs
Below is the list of APIs that still show up as undocumented in dotnet-api-docs and were introduced in .NET 9.

Full porting instructions can be found in the [main issue](https://github.com/dotnet/runtime/issues/105974).

This task needs to be finished **the day before the RC2 snap** (September 15th).

P7 Undoc APIs as of 08/20: https://github.com/dotnet/runtime/issues/105985#issuecomment-2303250806

<details><summary>Outdated</summary>

Scroll right to see the details:

| API                                                                                                                                                                                                        | Summary  | Parameters | TypeParameters | ReturnValue |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|------------|----------------|-------------|
| [M:System.Threading.Tasks.Task.WhenAll1(System.ReadOnlySpan{System.Threading.Tasks.Task{0}})](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Threading.Tasks/Task.xml)                     | Present  | Present    | Missing        | Present     |
| [M:System.Threading.Tasks.Task.WhenAny1(System.ReadOnlySpan{System.Threading.Tasks.Task{0}})](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Threading.Tasks/Task.xml)                     | Present  | Present    | Missing        | Present     |
| [M:System.Threading.Tasks.Task.WhenEach1(System.ReadOnlySpan{System.Threading.Tasks.Task{0}})](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Threading.Tasks/Task.xml)                    | Present  | Present    | Missing        | Present     |
| [M:System.Threading.Tasks.Task.WhenEach1(System.Threading.Tasks.Task{0}[])](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Threading.Tasks/Task.xml)                                       | Present  | Present    | Missing        | Present     |
| [M:System.Threading.Tasks.Task.WhenEach1(System.Collections.Generic.IEnumerable{System.Threading.Tasks.Task{0}})](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Threading.Tasks/Task.xml) | Present  | Present    | Missing        | Present     |

</details>

### PR DESCRIPTION
**Pull request by ericstj**
Fix https://github.com/dotnet/runtime/issues/105985

API-docs side of the change: https://github.com/dotnet/dotnet-api-docs/pull/10405

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs [hunk 1]
```diff
@@ -6837,25 +6844,33 @@ public static IAsyncEnumerable<Task> WhenEach(params Task[] tasks)
         }
 
         /// <inheritdoc cref="WhenEach(Task[])"/>
+        /// <param name="tasks">The task to iterate through when completed.</param>
         public static IAsyncEnumerable<Task> WhenEach(ReadOnlySpan<Task> tasks) => // TODO https://github.com/dotnet/runtime/issues/77873: Add params
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6848]: Oh no. We're missing params. This got missed.

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs [hunk 2]
```diff
@@ -6837,25 +6844,33 @@ public static IAsyncEnumerable<Task> WhenEach(params Task[] tasks)
         }
 
         /// <inheritdoc cref="WhenEach(Task[])"/>
+        /// <param name="tasks">The task to iterate through when completed.</param>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6847]:
```suggestion
        /// <param name="tasks">The tasks to iterate through as they complete.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs [hunk 3]
```diff
@@ -6837,25 +6844,33 @@ public static IAsyncEnumerable<Task> WhenEach(params Task[] tasks)
...
         public static IAsyncEnumerable<Task> WhenEach(ReadOnlySpan<Task> tasks) => // TODO https://github.com/dotnet/runtime/issues/77873: Add params
             WhenEachState.Iterate<Task>(WhenEachState.Create(tasks));
 
         /// <inheritdoc cref="WhenEach(Task[])"/>
+        /// <param name="tasks">The task to iterate through when completed.</param>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6852]:
```suggestion
        /// <param name="tasks">The tasks to iterate through as they complete.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs [hunk 4]
```diff
@@ -6837,25 +6844,33 @@ public static IAsyncEnumerable<Task> WhenEach(params Task[] tasks)
...
         public static IAsyncEnumerable<Task> WhenEach(IEnumerable<Task> tasks) =>
             WhenEachState.Iterate<Task>(WhenEachState.Create(tasks));
 
         /// <inheritdoc cref="WhenEach(Task[])"/>
+        /// <typeparam name="TResult">The type of the result returned by the tasks.</typeparam>
+        /// <param name="tasks">The task to iterate through when completed.</param>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6858]:
```suggestion
        /// <param name="tasks">The tasks to iterate through as they complete.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs [hunk 5]
```diff
@@ -6837,25 +6844,33 @@ public static IAsyncEnumerable<Task> WhenEach(params Task[] tasks)
...
         public static IAsyncEnumerable<Task<TResult>> WhenEach<TResult>(params Task<TResult>[] tasks)
         {
             ArgumentNullException.ThrowIfNull(tasks);
             return WhenEach((ReadOnlySpan<Task<TResult>>)tasks);
         }
 
         /// <inheritdoc cref="WhenEach(Task[])"/>
+        /// <typeparam name="TResult">The type of the result returned by the tasks.</typeparam>
+        /// <param name="tasks">The task to iterate through when completed.</param>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6867]:
```suggestion
        /// <param name="tasks">The tasks to iterate through as they complete.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs [hunk 6]
```diff
@@ -6837,25 +6844,33 @@ public static IAsyncEnumerable<Task> WhenEach(params Task[] tasks)
...
         public static IAsyncEnumerable<Task<TResult>> WhenEach<TResult>(ReadOnlySpan<Task<TResult>> tasks) => // TODO https://github.com/dotnet/runtime/issues/77873: Add params
             WhenEachState.Iterate<Task<TResult>>(WhenEachState.Create(ReadOnlySpan<Task>.CastUp(tasks)));
 
         /// <inheritdoc cref="WhenEach(Task[])"/>
+        /// <typeparam name="TResult">The type of the result returned by the tasks.</typeparam>
+        /// <param name="tasks">The task to iterate through when completed.</param>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6873]:
```suggestion
        /// <param name="tasks">The tasks to iterate through as they complete.</param>
```

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs (hunk 1 above)
Comment (ericstj) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6848]: I guess `<inheritdoc` omits the parameters if the type is different, even if the name is the same.

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6848]: I meant we have a product bug: the signature should be `WhenEach(params ReadOnlySpan<Task> tasks)`

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6848]: https://github.com/dotnet/runtime/pull/107964
Probably doesn't meet the bar for .NET 9 at this point, unfortunately.

Comment (carlossanlop) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6848]: Should we consider asking for an exception to fix the signature? I think we could even sneak it into RC2 if the branch is still open for the next day or two.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs (hunk 1 above)
Comment (ericstj) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6848]: I see, we should fix this.  Better to fix now before we ship it.

How many did we miss? https://github.com/dotnet/runtime/blob/5d8d04ec7b77299cdde6c3d069b0477f190cd121/src/libraries/System.Runtime/ref/System.Runtime.cs#L15884-L15889

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6848]: I'm aware of the two I fixed in the pr linked above. Still needs approval.

Comment (ericstj) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:6848]: I see from Stephen's PR, only 2 were missed: https://github.com/dotnet/runtime/pull/107964

I'm not inclined to do any heroics over this.  It's not breaking to add params.  We can just fix in 10.0

**Pull request merged**